homeassistant/components/light/knx.py

```diff
@@ -154,19 +233,57 @@ def supported_features(self):
         if self.device.supports_brightness:
             flags |= SUPPORT_BRIGHTNESS
         if self.device.supports_color:
-            flags |= SUPPORT_COLOR
+            flags |= SUPPORT_COLOR | SUPPORT_BRIGHTNESS
+        if self.device.supports_color_temperature or \
+                self.device.supports_tunable_white:
+            flags |= SUPPORT_COLOR_TEMP
         return flags
 
     async def async_turn_on(self, **kwargs):
         """Turn the light on."""
-        if ATTR_BRIGHTNESS in kwargs:
-            if self.device.supports_brightness:
-                await self.device.set_brightness(int(kwargs[ATTR_BRIGHTNESS]))
-        elif ATTR_HS_COLOR in kwargs:
-            if self.device.supports_color:
-                await self.device.set_color(color_util.color_hs_to_RGB(
-                    *kwargs[ATTR_HS_COLOR]))
+        brightness = kwargs.get(ATTR_BRIGHTNESS, self.brightness)
+        hs_color = kwargs.get(ATTR_HS_COLOR, self.hs_color)
+        mireds = kwargs.get(ATTR_COLOR_TEMP, self.color_temp)
+
+        # fall back to default values, if required
+        if brightness is None:
+            brightness = DEFAULT_BRIGHTNESS
+        if hs_color is None:
+            hs_color = DEFAULT_COLOR
+
+        update_brightness = ATTR_BRIGHTNESS in kwargs
+        update_color = ATTR_HS_COLOR in kwargs
+        update_color_temp = ATTR_COLOR_TEMP in kwargs
+
+        # always only go one path for turning on (avoid conflicting changes
+        # and weird effects)
+        if self.device.supports_brightness and \
+                (update_brightness and not update_color):
+            # if we don't need to update the color, try updating brightness
+            # directly if supported; don't do it if color also has to be
+            # changed, as RGB color implicitly sets the brightness as well
+            await self.device.set_brightness(brightness)
+        elif self.device.supports_color and \
+                (update_brightness or update_color):
+            # change RGB color (includes brightness)
+            await self.device.set_color(
+                color_util.color_hsv_to_RGB(*hs_color, brightness * 100 / 255))
+        elif self.device.supports_color_temperature and \
+                update_color_temp:
+            # change color temperature without ON telegram
+            kelvin = int(color_util.color_temperature_mired_to_kelvin(mireds))
+            if kelvin > self._max_kelvin:
+                kelvin = self._max_kelvin
+            elif kelvin < self._min_kelvin:
+                kelvin = self._min_kelvin
```

Comment:
I used your proposed approach in a previous version. It came to rounding errors when reaching _min_mireds. 
If I remember correctly: 
- User set min_kelvin to 2550 -> lowest possible value from HA-plugin to xknx after conversion and comparison in Mireds and conversion to Kelvin was 2551. 
- User set max_kelvin to 6200 -> highest value from HA-plugin to xknx after conversion and comparison in Mireds and conversion to Kelvin was 6211 which exceeds the user setting. 

I would like to leave the user defined temperature boundaries in Kelvin because it is the same Unit thats used in the KNX actuator parameterisation.

Reply:
We can't remove `self.min_mireds` and `self.max_mireds` propertys because they are needed to override the default properties of the `Light` class. We could remove the `self._min_mireds` attributes and put its calculation `color_util.color_temperature_kelvin_to_mired(self._max_kelvin)`
  in the `self.min_mireds` property.